src/tools/compiletest/src/lib.rs

```diff
@@ -1128,7 +1130,7 @@ fn not_a_digit(c: char) -> bool {
     !c.is_digit(10)
 }
 
-fn check_overlapping_tests(found_paths: &BTreeSet<PathBuf>) {
+fn check_overlapping_tests(found_paths: &HashSet<PathBuf>) {
```

Comment:
I think this can be significantly improved by avoiding multiple iterations.

Currently we do:

.loop over `found_paths` to generate `collisions` (first iteration)
...sort `collisions` (second iteration)
.....loop over `collisions` to build `String` value. (third iteration)

What I suggest instead:

. Build `String` value while during the loop of `found_paths` (we need to keep the `BTreeSet` for this)

Reply:
Ah, I didn't see that one. Thanks for clarification